dxr/plugins/python/indexers.py

```diff
@@ -187,54 +266,28 @@ def _all_needles(self):
             if tok_name in ('def', 'class'):
                 previous_start = start
             elif previous_start is not None:
-                node_start_table[previous_start] = start
+                self.node_start_table[previous_start] = start
                 previous_start = None
 
-        # Run through the AST looking for things to index!
-        syntax_tree = ast.parse(self.contents.encode('utf-8'))
-        for node in ast.walk(syntax_tree):
-            # Index classes and functions for the type: and function:
-            # filters.
-            if isinstance(node, ast.ClassDef) or isinstance(node, ast.FunctionDef):
-                node.start = (node.lineno, node.col_offset)
-                if node.start in node_start_table:
-                    node.start = node_start_table[node.start]
-
-                node.end = (node.start[0], node.start[1] + len(node.name))
-                needle_type = self.needle_types[node.__class__]
-                yield self._needle(needle_type, node.name, node.start, node.end)
-
-            # Index the class hierarchy for classes for the derived: and
-            # bases: filters.
-            if isinstance(node, ast.ClassDef):
-                class_name = module_name + '.' + node.name
-
-                bases = self.get_bases_for_class(class_name)
-                for qualname in bases:
-                    name = qualname.split('.')[-1]
-                    yield self._needle(needle_type='py_derived',
-                                       name=name, qualname=qualname,
-                                       start=node.start, end=node.end)
-
-                derived_classes = self.get_derived_for_class(class_name)
-                for qualname in derived_classes:
-                    name = qualname.split('.')[-1]
-                    yield self._needle(needle_type='py_bases',
-                                       name=name, qualname=qualname,
-                                       start=node.start, end=node.end)
-
-    def _needle(self, needle_type, name, start, end, qualname=None):
-        return (
-            needle_type,
-            {'name': name,
-             'qualname': qualname,
-             'start': start[1],
-             'end': end[1]},
-            Extent(Position(row=start[0],
-                            col=start[1]),
-                   Position(row=end[0],
-                            col=end[1]))
-        )
+
+    def get_node_start_end(self, node):
+        """Return start and end positions within the file for the given
+        AST Node.
+
+        """
+        start = (node.lineno, node.col_offset)
```

Comment:
You don't need the parens if you don't want. But if they make you happy, keep them. :-)